tools/cli/adminDBScanCommand.go

```diff
@@ -564,6 +587,48 @@ func verifyFirstHistoryEvent(
 	return VerificationResultNoCorruption
 }
 
+// Checks for validity of activity ids.
+// This refers to an accident when DB or our code wrote incorrect values that looked like some huge int64s.
+func verifyActivityIds(
+	shardID int32,
+	nextEventID int64,
+	activityInfos map[int64]*persistencespb.ActivityInfo,
+	executionInfo *persistencespb.WorkflowExecutionInfo,
+	executionState *persistencespb.WorkflowExecutionState,
+	corruptedExecutionWriter BufferedWriter,
+	branch *persistencespb.HistoryBranch,
+	) (
+	VerificationResult,
+) {
+	if (len(activityInfos) > 0) {
+		for activityId, _ := range activityInfos {
+			if activityId >= nextEventID {
+				byteBranch, err := byteKeyFromProto(branch)
+				if err != nil {
+					return VerificationResultCheckFailure
+				}
```

Comment:
lazy initialization?

Reply:
Can you elaborate please?

Reply:
`byteBranch, err := byteKeyFromProto(branch)` this function is invoked per `activityInfo` within `activityInfos`
the branch object does not change, so either do `byteBranch, err := byteKeyFromProto(branch)` at the very beginning or do lazy initialization